.github/workflows/build.yml

```diff
@@ -98,3 +98,38 @@ jobs:
       - name: Setup tmate debug session
         uses: mxschmitt/action-tmate@v3
         if: env.RUN_TMATE
+  test:
+    name: Run Go Package Tests
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Download Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '^1.20.2'
+        id: go
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+
+      - name: "Retrieve package location"
+        run: |
+            echo PACKAGE_LOCATION="$(find ./ -name 'test')/../" >> $GITHUB_OUTPUT
+        id: packagelocation
+
+      - name: Generate Mocks and Run Tests
+        working-directory: ${{ steps.packagelocation.outputs.PACKAGE_LOCATION }}
+        run: |
+          export GOPATH="$HOME/go/"
+          export PATH=$PATH:$GOPATH/bin
+          go install github.com/golang/mock/mockgen@v1.6.0
+          go install github.com/jstemmer/go-junit-report@latest
```

Comment:
Do we want to move the versions for `mock` and `go-junit-report` into https://github.com/cisagov/setup-env-github-action/blob/develop/action.yml, similar to what we do for `packer`, `terraform`, etc?  That would allow us to update these versions in one central location, as opposed to requiring a "Go Kraken" whenever these versions change.

Reply:
Will wait for additional input.

Reply:
My thumbs up was my signal that I like @dav3r's idea.

Reply:
oh I missed that. oops

Reply:
Please see commit ecbc0939d1f16bce6d60cf088943fd6904f530dc

Reply:
That looks good, but this won't work as intended until https://github.com/cisagov/setup-env-github-action/pull/63 has been merged.

Reply:
Now that https://github.com/cisagov/setup-env-github-action/pull/63 has been merged, I am marking this conversation as resolved.